jdaviz/configs/imviz/helper.py

```diff
@@ -209,6 +209,83 @@ def offset_by(self, dx, dy):
                 viewer.state.x_max = viewer.state.x_min + width
                 viewer.state.y_max = viewer.state.y_min + height
 
+    @property
+    def zoom_level(self):
+        """Zoom level:
+
+        * 1 means real-pixel-size.
```

Comment:
@mwcraig , what does "real pixel size" really mean?

Reply:
I interpreted that to me 1 pixel in the image was 1 pixel in the display. I'm handling it in bqplot viewer this way: https://github.com/astropy/astrowidgets/pull/142/files#diff-6aea2ce078dbeb19a51bdb4018c2b963ffa8728ce163c5d91aeb3bf42a3a2293R207

Reply:
Hmm... @astrofrog , you happen to know the best way to translate this low-level `bqplot` functionality to `jdaviz` way of doing stuff? `self._figure` here is the underlying `bqplot` `Figure` object.

```python
        figure_width = float(self._figure.layout.width[:-2])
        new_width = figure_width / zoom_level
        self.set_size(new_width, 'x')
        self._set_scale_aspect_ratio_to_match_viewer('y')
```

Reply:
Wait a min... @mwcraig , did you also completely changed the zooming API?

Reply:
I don't think so, no. I had already written a method to change the scales to ensure square pixels (took me way long than I should hace) so get the figure width, then divide by `zoom_level` to get the number of pixels from the image that should be displayed given that zoom level, set the width, then fix the aspect ratio. It is not real pretty but seems to work.

Reply:
Doesn't look like I can use your shortcut. For some reason, the equivalent of your `self._figure.layout` that is `viewer.figure.layout` here, returns a `None`. 😿

Reply:
I code dived here and there and couldn't find anything... maybe @maartenbreddels know?

Reply:
I had added the layout in my viewer.

Reply:
viewer.shape gives the width and height of the image in display coordinates, is that what you need?

Reply:
Thanks, that is what I needed. I think I fixed the math.